assets/app/scripts/controllers/modals/editModal.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:EditModalController
+ * @description
+ * # ProjectController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('EditModalController', function ($scope, $filter, $modalInstance, DataService) {
+    // Use angular.copy to avoid $$hashKey properties inserted by ng-repeat.
+    // TODO: Add apiVersion?
+    var resource = angular.copy($scope.resource);
+    if (resource.kind) {
+      // Hack to make `kind` appear at the top.
+      //
+      // Since `kind` is inserted by DataService for list operations, it is
+      // inserted last. yamljs serializes using Object.keys() ordering with no
+      // option to control order. Most browsers return keys in insertion order,
+      // however, so if we add kind first, it appears at the top of the
+      // serialized YAML. The rest of the properties are in the order returned
+      // from the API in the order we want (metadata, spec, status).
+      resource = angular.extend({ kind: resource.kind }, resource);
+    }
+
+    $scope.model = YAML.stringify(resource, 8, 2);
+
+    $scope.aceLoaded = function(editor) {
+      // Resize the editor based on window height.
+      var updateEditorHeight = function() {
+        var headerHeight = $('.modal-header').outerHeight();
+        var footerHeight = $('.modal-footer').outerHeight();
+        var availableHeight = window.innerHeight - headerHeight - footerHeight;
+
+        // Use 85% of available height.
+        var editorHeight = Math.floor(availableHeight * 0.85);
+
+        // Animate the change so it's not so janky.
+        // min-height set in CSS.
+        $('.editor').animate({
+          height: editorHeight + 'px'
+        }, 30, function() {
+          editor.resize();
+        });
+      };
+
+      setTimeout(updateEditorHeight, 10);
+
+      var onResize = _.debounce(updateEditorHeight, 200);
+      $(window).resize(onResize);
+      $scope.$on('$destroy', function() {
+        // Stop listening for resize events.
+        $(window).off('resize', onResize);
+      });
+    };
+
+    $scope.save = function() {
+      var updatedResource;
+      try {
+        updatedResource = YAML.parse($scope.model);
+      } catch (e) {
+        $scope.error = {
+          details: e.message || "Unable to parse YAML."
+        };
+        return;
+      }
+
```

Comment:
probably want a sanity check that the kind (and maybe the apiVersion) hasn't changed... trying to think through what happens if those get modified